### ISSUE TITLE
AIP-84 Refactor dataset dependencies graph

### ISSUE THREAD

**Issue by pierrejeambrun** [apache/airflow#42368]
Migrate the asset dependency private endpoint and also add additional features. This is a backend only change.

Showing all separate dataset dependencies graph is not useful. We only care about showing the full graph connected to a selected dag or dataset. Furthermore, connections dataset -> dag should render the dag's trigger conditions just like we did in for the dag graph.

Update:
Waiting on the front-end assets design to be finalized to know what features are exactly needed by the asset dependency graph private endpoint.

### Committer

- [X] I acknowledge that I am a maintainer/committer of the Apache Airflow project.

**Comment by bbovenzi** [comment]
We need to update `get_dag_dependencies()` on the serialized_dag model. Because right now it is a single dict of many disconnected graphs. Instead, it should be a list of graphs. Then we can pass a `node_id`, representing a dag or dataset, to return which graph contains that node.

**Comment by bbovenzi** [comment]
We can wait on this until we have finalized data assets UI designs.

**Comment by pierrejeambrun** [comment]
Putin this on hold. Let me know when I should work on it.

**Comment by pierrejeambrun** [comment]
Checked with Rahul.

I'll take this one.
